counsel.el

```diff
@@ -4918,6 +4918,36 @@ An extra action allows to switch to the process buffer."
 ;; `counsel-slime-repl-history' within
 ;; `counsel--browse-history-action'.
 
+;;** `counsel-esh-dir-history'
+(declare-function eshell/cd "eshell")
+
+(defun counsel--esh-dir-history-action-cd (pair)
+  "Default action for counsel-esh-dir-history. It changes the
+current working directory to the one selected by the user."
+  (eshell/cd (car pair)))
+
+(defun counsel--esh-dir-history-action-edit (pair)
+  "Action for counsel-esh-dir-history to insert the selected
+directory to the Eshell buffer prefixed by \"cd \", allowing the
+caller to modify parts of the directory before switching to it."
+  (ivy--action-insert (format "cd %s" (car pair))))
```

Comment:
Why is `ivy--action-insert` needed here? (As opposed to plain `insert`.)

Reply:
He, because of me being lazy. I wrongly assumed that using `ivy--action-insert` was kind of a convention inside ivy to insert but of course I didn't read the code to know when it'd be actually necessary :). It's changed.

Reply:
I think you'll soon find out that Ivy doesn't have conventions, and if it does, it doesn't follow them ;).